server/controller/challenge.js

```diff
@@ -0,0 +1,189 @@
+    difficulty: Joi.number().min(1).max(3).required(),
+    description: Joi.string().required(),
+    repo_url: Joi.string(),
+    assets: Joi.string(),
+    cover_photo: Joi.string(),
+  });
+
+  // Schema Validation
+  try {
+    await schema.validateAsync({
+      topic,
+      expertise,
+      difficulty,
+      description,
+      repo_url,
+      assets,
+      cover_photo,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      error: error.message,
+    });
+  }
+
+  try {
+    // Find an Expertise with topic [expertise] and auth0_ref [sub]
+    const expertise = expertise.findOne({ topic: expertise, auth0_ref: sub });
+    if (!expertise) {
+      return res.status(404).json({
+        msg: "Expertise not Found!",
+      });
+    }
+    let newChallenge = new challengeDB({
+      expertise_id: expertise._id,
+      topic,
+      difficulty,
+      description,
+      repo_url,
+      assets,
+      cover_photo,
+    });
+    await newChallenge.save();
+
+    return res.status(200).json({
+      msg: "Challenge Added",
+    });
+  } catch (error) {
+    return res.status(500).json({
+      error: "Server Error!",
+    });
+  }
+};
+
+exports.updateChallenge = async (req, res) => {
+  const {
+    challenge_id,
+    topic,
+    difficulty,
+    description,
+    repo_url,
+    assets,
+    cover_photo,
+  } = req.body;
+
+  const schema = Joi.object().keys({
+    challenge_id: Joi.string().required(),
+    topic: Joi.string().required(),
+    expertise: Joi.string().required(),
+    difficulty: Joi.number().min(1).max(3).required(),
+    description: Joi.string().required(),
+    repo_url: Joi.string(),
+    assets: Joi.string(),
+    cover_photo: Joi.string(),
+  });
+
+  // Schema Validation
+  try {
+    await schema.validateAsync({
+      topic,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      error: error.message,
+    });
+  }
+
+  try {
+    let challenge = await challengeDB.findOne({
+      _id: challenge_id,
+      auth0_ref: sub,
+    });
+
+    if (!challenge) {
+      return res.status(404).json({
+        msg: "Challenge not Found!",
+      });
+    }
+
+    challenge = {
+      ...challenge,
+      topic,
+      difficulty,
+      description,
+      repo_url,
+      assets,
+      cover_photo,
+    };
+    await challenge.save();
+
+    return res.status(200).json({
+      msg: "Challenge Updated",
+    });
+  } catch (error) {
+    return res.status(500).json({
+      error: "Server Error!",
+    });
+  }
+};
+
+exports.removeChallenge = async (req, res) => {
+  const { challenge_id } = req.body;
+
+  const schema = Joi.object().keys({
+    challenge_id: Joi.string().required(),
+  });
+
+  // Schema Validation
+  try {
+    await schema.validateAsync({
+      challenge_id,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      error: error.message,
+    });
+  }
+
+  try {
+    let challenge = learningDB.findById(challenge_id);
```

Comment:
convert challenge id to object id i think

Reply:
Mongoose converts it automatically.